src/entities.rs

```diff
@@ -29,17 +31,21 @@ impl Entity {
     /// `World::spawn_at` for easy serialization. Alternatively, consider `id` for more compact
     /// representation.
     pub fn to_bits(self) -> u64 {
-        u64::from(self.generation) << 32 | u64::from(self.id)
+        NonZeroU64::from(self.generation).get() << 32 | u64::from(self.id)
     }
 
     /// Reconstruct an `Entity` previously destructured with `to_bits`
     ///
     /// Useful for storing entity IDs externally, or in conjunction with `Entity::to_bits` and
     /// `World::spawn_at` for easy serialization.
```

Comment:
We should document the conditions in which `None` is returned, i.e. when `bits` does not correspond to a possible return value from `to_bits`.